Realm/Realm.modulemap

```diff
@@ -18,8 +18,6 @@ framework module Realm {
         header "RLMRealm_Private.h"
         header "RLMResults_Private.h"
         header "RLMSchema_Private.h"
-        header "RLMSyncConfiguration_Private.h"
-        header "RLMSyncUtil_Private.h"
```

Comment:
It might make sense to declare here `config_macros REALM_ENABLE_SYNC` (as seen in the [Clang docs](https://clang.llvm.org/docs/Modules.html#configuration-macros-declaration)), so that the compiler knows that the Clang module API looks different on base of that and can use that to invalidate the cache. That might help to avoid problems where derived data would have to be cleared when switching from one variant to the other.